dashboard/src/components/Catalog/CatalogItem.tsx

```diff
@@ -45,15 +46,18 @@ const CatalogItem: React.SFC<ICatalogItemProps> = props => {
   let subIcon;
   if (type === "chart") {
     tag1 = (
+      // TODO (absoludity): It appears that repo is an optional prop only because the item
+      // may be an operator. We could instead use polymorphism with two separate
+      // types ICatalogItemChart and ICatalogItemOperator which both implement
+      // a required interface for use in the component?
```

Comment:
interfaces/inheritance are discouraged in react (if I understood your comment).

What we could do is move the link/icon resolution to the caller in where we are sure that for example, `repo` is populated for charts and remove this `if/else` from here.

Reply:
> interfaces/inheritance are discouraged in react (if I understood your comment).

I don't think interfaces are discouraged? What you normally want to discourage is indeed inheritance (ie. prefer composition over inheritance) but I didn't mention inheritance here - it'd be composition.
 
> What we could do is move the link/icon resolution to the caller in where we are sure that for example, `repo` is populated for charts and remove this `if/else` from here.

We could - but not sure it's neater, it just avoids dealing with the fact that we're representing two different things (charts and operators) with a single implementation and if/else's.

Anyway, let me move the comment to an issue so we can discuss there (created #1803). I may take a look at it tomorrow, as it's bitten me a couple of times.